### PR TITLE
UX - Add explanations about colours in the table

### PR DESCRIPTION
**Pull request by Gustry**
Follow up from #3418 

image

* [x] Format versions
* [ ] Add a span with colour for visual aspect
* [x] Make strings translatable
* [x] Rephrase

### CODE REVIEW

**Comment by Gustry** [comment]
image


All rules added, with a blocking rules versus not blocking.

@nworr Do you think you can contribute by adding a plugin template to format versions ? https://docs.jelix.org/fr/manuel/plugins/tpl Similar to what you did in #3449 It would be nice if you can contribute on this PR ? 😊

Side request as well @nworr Is-it possible to add `<strong>%s</strong>` in a `.properties` files ? I prefer the rendering of my first screenshot with these dynamic values in bold.

**Comment by nworr** [comment]
@Gustry i add modifier : `qgis_majmin_human_version` , but for the locale you may use the function `jtpl_modifier_common_qgis_majmin_human_version`

You can use html tags in the locale, but not sure it's the best way

**Comment by Gustry** [comment]
Ready for review : 

image

@mdouchin Can you check the strings ?

Maybe we can come back if we really want to add the last bullet point about the colors ?

File: lizmap/modules/admin/templates/project_list_zone.tpl [hunk 1]
```diff
@@ -222,3 +234,58 @@
     {/foreach}
     </tbody>
 </table>
+
+<div>
+    <br>
+    <b>{@admin.project.rules.list.introduction@}</b>
```
Comment (Gustry) [inline lizmap/modules/admin/templates/project_list_zone.tpl:240]:
```suggestion
    <strong>{@admin.project.rules.list.introduction@}</strong>
```

File: lizmap/modules/admin/zones/project_list.zone.php [hunk 2]
```diff
@@ -83,20 +83,41 @@ protected function _prepareTpl()
...
             // QGIS server
             $qgisServerVersion = $data['qgis_server_info']['metadata']['version'];
             $serverVersions['qgis_server_version'] = $qgisServerVersion;
             $explode = explode('.', $qgisServerVersion);
             // Keep only major and minor version
-            $serverVersions['qgis_server_version_int'] = (int) $explode[0].str_pad($explode[1], 2, '0', STR_PAD_LEFT);
-
+            $qgisServerVersionInt = intval($explode[0].str_pad($explode[1], 2, '0', STR_PAD_LEFT));
+            $serverVersions['qgis_server_version_int'] = $qgisServerVersionInt;
+            $serverVersions['qgis_server_version_humanreadable'] = $this->qgisMajMinHumanVersion($qgisServerVersionInt);
```
Comment (Gustry) [inline lizmap/modules/admin/zones/project_list.zone.php:95]:
```suggestion
            $serverVersions['qgis_server_version_human_readable'] = $this->qgisMajMinHumanVersion($qgisServerVersionInt);
```

File: lizmap/modules/admin/templates/project_list_zone.tpl [hunk 3]
```diff
@@ -222,3 +233,56 @@
     {/foreach}
     </tbody>
 </table>
+
+<div>
+    <br>
+    <strong>{@admin.project.rules.list.introduction@}</strong>
+    <p>
+        {jlocale "admin.project.rules.list.description.html", array($lizmapVersion, ($serverVersions['qgis_server_version_humanreadable']))}
```
Comment (Gustry) [inline lizmap/modules/admin/templates/project_list_zone.tpl:241]:
```suggestion
        {jlocale "admin.project.rules.list.description.html", array($lizmapVersion, ($serverVersions['qgis_server_version_human_readable']))}
```

File: lizmap/modules/admin/locales/en_US/admin.UTF-8.properties [hunk 4]
```diff
@@ -273,8 +273,25 @@ server.information.qgis.update=QGIS Server needs to be updated at least to versi
 server.information.plugin.update=The %s plugin needs to be updated.
 
 
-project.not.displayed=Some projects, listed below, can not be displayed in the main web interface because they need an update from the GIS administrator.
-project.minimum.target.version.required=A QGIS project must be designed for at least Lizmap Web Client %s to be displayed in this current version of Lizmap Web Client
+server.information.error=Before checking your projects, you must check your QGIS server settings in the "server information" panel. The configuration is currently not correct about QGIS Server.
```
Comment (mdouchin) [inline lizmap/modules/admin/locales/en_US/admin.UTF-8.properties:276]:
```suggestion
server.information.error=Before checking your projects, you must check your QGIS server settings in the "Server information" menu. The configuration of QGIS Server is currently not correct.
```

File: lizmap/modules/admin/locales/en_US/admin.UTF-8.properties [hunk 5]
```diff
@@ -273,8 +273,25 @@ server.information.qgis.update=QGIS Server needs to be updated at least to versi
...
+project.rules.list.target.version.html=Light yellow if the project is designed for Lizmap Web Client <strong>%s</strong>.
+project.rules.list.important.count.layers.html=Light yellow if the count is greater or equal to <strong>%s</strong> layers.
+project.rules.list.very.important.count.layers.html=Light coral if the count is greater or equal to <strong>%s</strong> layers.
+project.rules.list.custom.projection=Light coral if the project is using a <strong>user defined</strong> projection.
+project.rules.list.invalid.datasource.html=Light coral if the project has some <strong>invalid layer datasource</strong> when QGIS server loaded the project.
+project.rules.list.warning.loading.html=Light yellow if the project takes longer than <strong>%s</strong> seconds to load.
+project.rules.list.error.loading.html=Light coral if the project takes longer than <strong>%s</strong> seconds to load.
+project.rules.list.warning.memory.html=Light yellow if the project takes more than <strong>%s</strong> Mo to load.
```
Comment (mdouchin) [inline lizmap/modules/admin/locales/en_US/admin.UTF-8.properties:290]:
```suggestion
project.rules.list.warning.memory.html=Light yellow if the project takes more than <strong>%s</strong> Mo of the server memory to load.
```

File: lizmap/modules/admin/locales/en_US/admin.UTF-8.properties [hunk 6]
```diff
@@ -273,8 +273,25 @@ server.information.qgis.update=QGIS Server needs to be updated at least to versi
...
+project.rules.list.error.memory.html=Light coral if the project takes more than <strong>%s</strong> Mo to load.
```
Comment (mdouchin) [inline lizmap/modules/admin/locales/en_US/admin.UTF-8.properties:291]:
```suggestion
project.rules.list.error.memory.html=Light coral if the project takes more than <strong>%s</strong> Mo of the server memory to load.
```

File: lizmap/modules/admin/locales/en_US/admin.UTF-8.properties [hunk 7]
```diff
@@ -273,8 +273,25 @@ server.information.qgis.update=QGIS Server needs to be updated at least to versi
...
+project.rules.list.warnings.html=<strong>Warnings</strong>, projects are still visible, but you should have a look to these projects :
+project.rules.list.qgis.version.light.yellow.html=Light yellow if the QGIS project version is too old compared to the current QGIS Server version. Current value less or equal to <strong>%s</strong>.
+project.rules.list.qgis.version.light.coral.html=Light coral if the QGIS desktop version <strong>is above</strong> QGIS server. Current value greater or equal than <strong>%s</strong>.
+project.rules.list.target.version.html=Light yellow if the project is designed for Lizmap Web Client <strong>%s</strong>.
+project.rules.list.important.count.layers.html=Light yellow if the count is greater or equal to <strong>%s</strong> layers.
+project.rules.list.very.important.count.layers.html=Light coral if the count is greater or equal to <strong>%s</strong> layers.
+project.rules.list.custom.projection=Light coral if the project is using a <strong>user defined</strong> projection.
```
Comment (mdouchin) [inline lizmap/modules/admin/locales/en_US/admin.UTF-8.properties:286]:
```suggestion
project.rules.list.custom.projection=Light coral if the project is using a <strong>user-defined</strong> projection.
```

**Review by mdouchin — CHANGES_REQUESTED** [review]
LGTM, some minor changes requested

**Pull request merged**